source_manager.go

```diff
@@ -141,15 +158,121 @@ func NewSourceManager(an ProjectAnalyzer, cachedir string) (*SourceMgr, error) {
 		}
 	}
 
-	return &SourceMgr{
+	sm := &SourceMgr{
 		cachedir: cachedir,
 		lf:       fi,
 		srcs:     make(map[string]source),
 		srcfuts:  make(map[string]*unifiedFuture),
 		an:       an,
 		dxt:      pathDeducerTrie(),
 		rootxt:   newProjectRootTrie(),
-	}, nil
+		qch:      make(chan struct{}),
+	}
+
+	return sm, nil
+}
+
+// SetUpSigHandling sets up typical signal handling for a SourceMgr. It will
+// register a signal handler to be notified on:
+//
+//  - syscall.SIGINT
+//  - syscall.SIGHUP
+//  - syscall.SIGTERM
+//  - syscall.SIGQUIT
+//  - os.Interrupt
+func SetUpSigHandling(sm *SourceMgr) {
+	sigch := make(chan os.Signal)
```

Comment:
In an earlier version of this PR, I had additional controls in place to deal with multiple signals coming in (and even potentially interleaving), and the channel was buffered. I got rid of it, though, after reading around a bit and getting the impression that kernels generally prevent retransmission of a signal if that same signal is already being handled. And, in the current implementation, it shouldn't be possible for a second signal to ever be processed (I could maybe even do away with the for loop), so I dropped the buffer.

However, [looking again](http://man7.org/linux/man-pages/man7/signal.7.html), that view of kernel/signal behavior seems like, at best, a vast oversimplification. So, maybe I will reintroduce that logic, as well as that buffer.